### ISSUE TITLE
Hygieia API-UI connectivity issue

### ISSUE THREAD

**Issue by Rajeswa** [capitalone/Hygieia#1902]
I am facing issue in API - UI connectivity. API instance is running and am getting the response (response as true) when I hit http://localhost:8081/api/ping. I have deployed UI via gulp serve command and in the UI page its showing as no API connectivity. When I try to hit http://localhost:8081/api/dashboard, am getting empty json. From the API logs, I saw it has established connection to mongodb successfully. Can you help me to sort out this?

I have built the entire project successfully that I have cloned from GitHub. Mongodb,npm,bower installation -> successful. No error message in api log.

**command used:**

java -jar api.jar --spring.config.name=api --spring.config.location=/home/linux/.jenkins/jobs/Dashboard/workspace/api/api.properties

**api.properties:**
dbname=dashboarddb
dbusername=dashboarduser
dbpassword=dbpassword
dbhost=localhost
dbport=27017
server.contextPath=/api
server.port=8081
version.number=@application.version.number@

GET method http://localhost:3000/api/authenticationProviders & http://localhost:3000/api/appinfo returns 404 page not found response. But am getting response in localhost:8081/api/authenticationProviders as ['Standard'].

**Comment by matthewbeasley** [comment]
@Rajeswa I believe this is an issue with CORS handling within the api. Will you open up the console in your browser and post a screenshot so that I can confirm we are experiencing the same issue? Just open your browsers console and navigate to the page that shows no API connectivity.

**Comment by Rajeswa** [comment]
@matthewbeasley - Thanks for the support. I have found the mistake being made. As am running the spring boot server in 8081, I have made the port change in gulpfile.js from 8080(default port) to 8081 and rebuilt the UI code. Now its successfully establishing the connections between API & UI.

But when I started by Jenkins collector, its somehow getting redirected to localhost. I am having Hygieia being installed in 1 server and its collecting details from Jenkins installed in another 1 server. 

application.property file has below entry for Jenkins server:

jenkins.servers[0]=**http://10.236.110.81:8080/jenkins**

2017-12-15 01:38:00,493 INFO c.c.d.collector.CollectorTask - **http://10.236.110.81:8080/jenkins**
2017-12-15 01:38:00,493 INFO c.c.d.collector.CollectorTask - -----------------------------------
2017-12-15 01:38:00,580 INFO c.c.d.collector.DefaultHudsonClient - Fetching jobs 0/40 pageSize 1000...
2017-12-15 01:38:03,547 INFO c.c.d.collector.CollectorTask - Fetched jobs 3s
2017-12-15 01:38:03,547 INFO c.c.d.collector.CollectorTask - New jobs 0 0s
2017-12-15 01:38:03,665 WARN c.c.d.collector.DefaultHudsonClient - Credentials for the following url was not found. This could happen if the domain/subdomain/IP address in the build url returned by Jenkins and the Jenkins instance url in your Hygieia configuration do not match: "**http://localhost:8080/jenkins**/job/.NET_Build/1/api/json?tree=number,url,timestamp,duration,building,result,culprits[fullName],changeSets[items[user,author[fullName],revision,id,msg,timestamp,date,paths[file]],kind],changeSet[items[user,author[fullName],revision,id,msg,timestamp,date,paths[file]],kind,revisions[module,revision]],actions[lastBuiltRevision[SHA1,branch[SHA1,name]],remoteUrls]"
